pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -914,6 +916,46 @@ func (aws *AWSCloud) List(filter string) ([]string, error) {
 	return aws.getInstancesByRegex(filter)
 }
 
+// getAllZones retrieves  a list of all the zones in which nodes are running
+// It currently involves querying all instances
```

Comment:
I don't think creating PDs is a terribly common operation.  We used to list all instances every 10 seconds (but we no longer do that) so I think we should be OK.  Originally I did try caching, but now that we no longer poll Instances List, that becomes much harder.

I propose that we run this as-is for now, and then if/when we identify it as a problem we can add time-based caching.

Ideally the cloudprovider should have access to a watched set of Nodes, in which case this would be a trivial operation. This actually comes up in a number of places (e.g. LoadBalancer).  Maybe we do that in 1.4?